synapseclient/api/annotations.py

```diff
@@ -47,3 +47,36 @@ def set_annotations(
             }
         ),
     )
+
+
+async def set_annotations_async(
+    annotations: "Annotations",
+    synapse_client: Optional["Synapse"] = None,
+):
+    """Call to synapse and set the annotations for the given input.
+
+    Arguments:
+        annotations: The annotations to set. This is expected to have the id, etag,
+            and annotations filled in.
+        synapse_client: If not passed in or None this will use the last client from
+            the `.login()` method.
+
+    Returns: The annotations set in Synapse.
+    """
+    annotations_dict = asdict(annotations)
+
+    synapse_annotations = _convert_to_annotations_list(
+        annotations_dict["annotations"] or {}
+    )
+    from synapseclient import Synapse
```

Comment:
why here the import statement is in the function call?

Reply:
This is to prevent circular import issues. This is a "lazy" import. It was the easiest way to fix the problem, but I'd be open to suggestions.

Reply:
Thanks. I actually have never heard of lazy import before. I looked it up and it seems like the benefit of lazy import is to load a module when the module is being used. But`Synapse` is so commonly used which makes me wonder how big the benefit is.. But thanks for teaching me lazy import, I learned something new today!

Reply:
@linglp to circle back around to this.  It's important to note that we use lazy importing to avoid circular imports.